onnxruntime/python/tools/quantization/data_reader.py

```diff
@@ -0,0 +1,495 @@
+import random
+import sys
+import time
+import torch
+
+# Setup logging level to WARN. Change it accordingly
+logging.basicConfig(format = '%(asctime)s - %(levelname)s - %(name)s -   %(message)s',
+                    datefmt = '%m/%d/%Y %H:%M:%S',
+                    level = logging.INFO)
+
+logging.getLogger("transformers.modeling_utils").setLevel(
+   logging.INFO)  # Reduce logging
+
+logger = logging.getLogger(__name__)
+
+
+def parse_annotations(filename):
+    import json
+    annotations = {}
+    with open(filename, 'r') as f:
+        annotations = json.load(f)
+
+
+    img_name_to_img_id = {}
+    for image in annotations["images"]:
+        file_name = image["file_name"]
+        img_name_to_img_id[file_name] = image["id"]
+
+    return img_name_to_img_id
+
+class ObejctDetectionDataReader(CalibrationDataReader):
+    def __init__(self, model_path='augmented_model.onnx'):
+        self.model_path = model_path
+        self.preprocess_flag = None
+        self.start_index = None
+        self.batch_size = None
+        self.size_limit = None
+        self.support_batch = False
+        self.batches = []
+
+    def set_start_index(self, i):
+        self.start_index = i
+
+    def set_size_limit(self, limit):
+        self.size_limit = limit
+
+    def set_batch_size(self, batch_size):
+        self.batches = []
+        self.batch_size = batch_size
+
+    def get_batch_size(self):
+        return self.batch_size
+
+    def set_preprocess_flag(self, flag):
+        self.preprocess_flag = flag
+
+class YoloV3DataReader(ObejctDetectionDataReader):
+    def __init__(self, calibration_image_folder,
+                       width=416,
+                       height=416,
+                       start_index=0,
+                       size_limit=0,
+                       batch_size=0,
+                       model_path='augmented_model.onnx',
+                       is_evaluation=False,
+                       annotations='./annotations/instances_val2017.json'):
+        ObejctDetectionDataReader.__init__(self, model_path)
+        self.image_folder = calibration_image_folder
+        self.model_path = model_path
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.datasize = 0
+        self.width = width
+        self.height = height
+        self.start_index = start_index
+        self.size_limit = size_limit
+        self.batch_size = batch_size
+        self.is_evaluation = is_evaluation
+        self.annotations = annotations
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            session = onnxruntime.InferenceSession(self.model_path, providers=['CPUExecutionProvider'])
+            print(session.get_inputs()[0].shape)
+            width = self.width 
+            height = self.width 
+            nchw_data_list, filename_list, image_size_list = yolov3_preprocess_func(self.image_folder, height, width, self.start_index, self.size_limit)
```

Comment:
you can load batch of images here, and keep the status of start_index. that's the reason why i don't think you need get_batch.

Reply:
Yes, I've modified the code here.